codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderConstraintViolations.kt

```diff
@@ -69,8 +69,11 @@ class ServerBuilderConstraintViolations(
         writer.rustBlock("pub${ if (visibility == Visibility.PUBCRATE) " (crate) " else "" } enum $constraintViolationSymbolName") {
             renderConstraintViolations(writer)
         }
-        renderImplDisplayConstraintViolation(writer)
-        writer.rust("impl #T for ConstraintViolation { }", RuntimeType.StdError)
+
+        if (all.isNotEmpty()) {
```

Comment:
Actually, this makes me realize the current implementation is wrong. It should `check` in the beginning of `render` that `all` is not empty and crash if it is. We don't want to render the constraint violation struct at all if no members are constrained.

Reply:
The rest of the changes in this file should not be needed then.